### DISCUSSION TITLE
Does this pin just control voltage on and off or is it a PWM pin?

### DISCUSSION THREAD

**Discussion by DIYCharles** [gigibu5/OctoLight#21] [Ideas]
For the pin does it just supply voltage to the pin like you would wire up a single led with a resistor. Or does it supply a pwm signal so you can run a transistor as a switch to control a much larger lighting circuit. If it does not I'd be happy to contribute so it can do so. Also with a pwm output not only can you control a much more power hungry light circuit you could build into the UI to set the brightness level.

**Comment by DIYCharles** [comment]
I see now it is using GPIO out HIGH and LOW at 3.3V level so I am going to make a commit to add some PWM settings for this to see if I can get that to work. Additionally this could be a cool framework to add some lighting notification actions using the code as a base.

**Comment by gigibu5** [comment]
The Raspberry Pi has 2 PWM capable pins, that are used for sound. You can use them for something else. This plugins only triggers the pins digitally (either 0V or 5V), currently it does not support PWM. Right now I do not have a lot of time on my hands, if you would like to implement it, you can create a pull request.
